tests/Integration/Infrastructure/Controller/LandingControllerTest.php

```diff
@@ -17,4 +16,15 @@ public function testList(): void
         $this->assertSame('Faciliter et sécuriser la circulation', $crawler->filter('h1')->text());
         $this->assertSame('Se connecter', $crawler->filter('div.user ul')->filter('li')->eq(0)->text()); // Logout user
     }
+
+    public function testLandingWithLoggedUser(): void
+    {
+        $client = $this->login();
+        $crawler = $client->request('GET', '/');
+
+        $this->assertResponseStatusCodeSame(200);
+        $this->assertSame('Faciliter et sécuriser la circulation', $crawler->filter('h1')->text());
+        $this->assertSame('Mathieu MARCHOIS', $crawler->filter('div.user ul')->filter('li')->eq(0)->text());
+        $this->assertSame('Accéder aux réglementations', $crawler->filter('a.fr-btn.fr-mr-2w')->text());
```

Comment:
Je suis peut-être relou avec cette idée (dans le passé on ne l'avait pas appliquée que ça n'a pas l'air de te convaincre) mais est-ce que ça ne serait pas plus lisible et robuste comme ça ?

```suggestion
        $enterLink = $crawler->filter('[data-testid="enter-link"]');
        $this->assertSame('Accéder aux réglementations', enterLink->text());
        $this->assertSame('/regulations', enterLink->attr('href'));
```

Avec `<a data-testid="enter-link" href="..." />` dans `index.html.twig`.

On peut utiliser le même `data-testid` dans les 2 branches (connecté / pas connecté) de sorte qu'on puisse s'assurer que seule la version attendue du bouton est affichée. (On voit que si on passait par des classes et qu'il se trouvait que les 2 versions du bouton étaient stylées différemment, ce serait plus difficile à faire. Là on fait plus comme un lecteur d'écran : "trouve moi l'élément qui s'appelle `"entry-link"`, quelle que soit sa forme")

Et ajouter une assertion similaire sur href dans testLanding ?

Reply:
Done